### PR TITLE
Provide the Kafka quickstart guide

### PR DESCRIPTION
**Pull request by cescoffier**
We had the code but not the writeup. This PR provides the instructions to build the Kafka example yourself.

### CODE REVIEW

File: docs/src/main/asciidoc/kafka-guide.adoc [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+    @Outgoing("topic-price")                            // <1>
+    public Flowable<Integer> generate() {               // <2>
+        return Flowable.interval(5, TimeUnit.SECONDS)
+                .map(tick -> random.nextInt(100));
+    }
+
+}
+----
+<1> Instruct Reactive Messaging to dispatch the items from returned stream to `topic-price`.
+<2> The method returns a RX Java 2 _stream_ (`Flowable`) emitting a random _prices_ every 5 seconds.
```
Comment (gsmet) [inline docs/src/main/asciidoc/kafka-guide.adoc]: s/prices/price/ (or drop the `a`)

File: docs/src/main/asciidoc/kafka-guide.adoc [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+
+This bean is generating random prices every 5 seconds. The method returns a _Reactive Streams_.
```
Comment (gsmet) [inline docs/src/main/asciidoc/kafka-guide.adoc]: You're saying exactly the same thing as above.

File: docs/src/main/asciidoc/kafka-guide.adoc [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+The generated items are sent to the stream named `topic-price`.
+This stream is mapped to Kafka using the `application.properties` file that we will create soon.
+
+== The price converter
+
+The price converter reads the prices from Kafka, and transforms them.
+Creates the `src/main/java/org/acme/quarkus/sample/PriceConverter.java` file with the following content:
```
Comment (gsmet) [inline docs/src/main/asciidoc/kafka-guide.adoc]: s/Creates/Create/

File: docs/src/main/asciidoc/kafka-guide.adoc [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+/**
+ * A bean producing random prices every 5 seconds.
+ * The prices are written to a Kafka topic (prices). The Kafka configuration is specified in the application configuration.
+ */
+@ApplicationScoped
+public class PriceGenerator {
+
+    private Random random = new Random();
+
+    @Outgoing("topic-price")                            // <1>
```
Comment (gsmet) [inline docs/src/main/asciidoc/kafka-guide.adoc]: Hmm, there's something weird with the name of topic. You're mentioning `prices` above. And the `@Incoming` also later references `prices` but it's named `topic-price` here? (note that I don't know Kafka at all).

Update: so now that I read the config stuff, I understand better. But it's a bit hard to follow at the beginning.

File: docs/src/main/asciidoc/kafka-guide.adoc [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+    @Incoming("prices")                                 // <1>
+    @Outgoing("my-data-stream")                         // <2>
+    @Broadcast                                          // <3>
+    public double process(int priceInUsd) {
+        return priceInUsd * CONVERSION_RATE;
+    }
+
+}
+----
+<1> Indicates that the method consumes the items from the `prices` stream
```
Comment (gsmet) [inline docs/src/main/asciidoc/kafka-guide.adoc]: It was referred to as a `topic` before, not a stream

File: docs/src/main/asciidoc/kafka-guide.adoc [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+import org.reactivestreams.Publisher;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+/**
+ * A simple resource retrieving the "in-memory" "my-data-stream" and sending the items to a server sent event.
```
Comment (gsmet) [inline docs/src/main/asciidoc/kafka-guide.adoc]: why double quotes around `"in-memory"`?

File: docs/src/main/asciidoc/kafka-guide.adoc [hunk 7]
```diff
@@ -0,0 +1,304 @@
...
+    @Inject
+    @Stream("my-data-stream") Publisher<Double> prices; // <1>
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public String hello() {
+        return "hello";
+    }
+
+
```
Comment (gsmet) [inline docs/src/main/asciidoc/kafka-guide.adoc]: Minor nitpick but one too many new lines

File: docs/src/main/asciidoc/kafka-guide.adoc [hunk 8]
```diff
@@ -0,0 +1,304 @@
...
+public class PriceResource {
+
+    @Inject
+    @Stream("my-data-stream") Publisher<Double> prices; // <1>
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public String hello() {
+        return "hello";
+    }
```
Comment (gsmet) [inline docs/src/main/asciidoc/kafka-guide.adoc]: Maybe you can drop this one?

File: docs/src/main/asciidoc/kafka-guide.adoc [hunk 9]
```diff
@@ -0,0 +1,304 @@
...
+}
+----
+<1> Injects the `my-data-stream` stream using the `@Stream` qualifier
+<2> Indicates that the content is sent using `Server Sent Events`
+<3> Returns the stream (_Reactive Stream_)
+
+== Configuring the Kafka connector
+
+We need to configure Kafka connector. This is done in the `application.properties` file.
+The keys are structured as follow:
```
Comment (gsmet) [inline docs/src/main/asciidoc/kafka-guide.adoc]: follows?

File: docs/src/main/asciidoc/kafka-guide.adoc [hunk 10]
```diff
@@ -0,0 +1,304 @@
...
+
+`smallrye.messaging.[sink|source].{stream-name}.property=value`
```
Comment (gsmet) [inline docs/src/main/asciidoc/kafka-guide.adoc]: We will really need to talk about if we want to achieve consistency with every configuration setting being `quarkus.<something>`. Maybe it doesn't make sense for MP components, I don't know, but I thought it was the convention.

(General comment, don't change anything)

File: docs/src/main/asciidoc/kafka-guide.adoc [hunk 11]
```diff
@@ -0,0 +1,304 @@
...
+
+The `stream-name` segment must match the value set in the `@Incoming` and `@Outgoing` annotation:
+* `topic-price` -> sink in which we write the prices
+* `prices` -> source in which we read the prices
```
Comment (gsmet) [inline docs/src/main/asciidoc/kafka-guide.adoc]: And here is what I was looking for :).

File: docs/src/main/asciidoc/kafka-guide.adoc [hunk 12]
```diff
@@ -0,0 +1,304 @@
...
+----
+
+Nothing spectacular here. On each received price, it updates the page.
+
+== Get it running
+
+If you followed the instructions, you should have Kafka running.
+Then, you just need to run the application using:
+
+`mvn compile quarkus:dev`
```
Comment (gsmet) [inline docs/src/main/asciidoc/kafka-guide.adoc]: Make it a proper `[source]`, it will look better.

File: docs/src/main/asciidoc/kafka-guide.adoc [hunk 13]
```diff
@@ -0,0 +1,304 @@
...
+
+Open `http://localhost:8080/prices.html` in your browser.
+
+NOTE: If you started the Kafka broker with docker compose, stop it using `CTRL+C` followed by `docker-compose down`.
+
+== Running Native
+
+You can build the native executable with:
+
+`mvn package -Pnative`
```
Comment (gsmet) [inline docs/src/main/asciidoc/kafka-guide.adoc]: Same here about using a `[source]`.

**Review by gsmet — CHANGES_REQUESTED** [review]
@cescoffier interesting one! I added some comments inline.

**Comment by gsmet** [comment]
@cescoffier I added the `noteworthy-feature` tag as I think new guides should be highlighted in the releases.

File: docs/src/main/asciidoc/kafka-guide.adoc [hunk 14]
```diff
@@ -0,0 +1,304 @@
...
+
+[source, subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=kafka-quickstart \
+    -Dextensions="reactive-kafka,vert.x"
+----
+
+This command generates a Maven project, importing the Reactive Messaging and Kafka connector extensions. Eclipse Vert.x is also required as the Kafka connector relies on Vert.x.
```
[the remark this answers is not in the document]
Comment (cescoffier) [reply docs/src/main/asciidoc/kafka-guide.adoc]: For some strange reason, it does not work when building the native image. It's a known bug, we need to have a look.

File: docs/src/main/asciidoc/kafka-guide.adoc (hunk 4 above)
Comment (cescoffier) [reply docs/src/main/asciidoc/kafka-guide.adoc]: I've changed `topic-price`. Unfortunately, it cannot be `price` as it would highjack the kafka part (it will just connect both together).

File: docs/src/main/asciidoc/kafka-guide.adoc (hunk 10 above)
Comment (cescoffier) [reply docs/src/main/asciidoc/kafka-guide.adoc]: This is still in discussion for the spec, so likely be changed soon. But yes, the question is whether we want `quarkus.`. The implementation would support this (you just need to add a bean in the extension).

**Review by gsmet — APPROVED** [review]
